src/player_comptask.c

```diff
@@ -1805,8 +1834,9 @@ short tool_dig_to_pos2_f(struct Computer2 * comp, struct ComputerDig * cdig, TbB
         digstl_y = stl_num_decode_y(i);
         digslb_x = subtile_slab(digstl_x);
         digslb_y = subtile_slab(digstl_y);
-        slb = get_slabmap_block(digslb_x, digslb_y);
-        if (slab_kind_is_liquid(slb->kind) && (computer_check_room_available(comp, RoK_BRIDGE) == IAvail_Now))
+        action_slb = get_slabmap_block(digslb_x, digslb_y);
+        if (slab_kind_is_liquid(action_slb->kind) &&  ((action_slb->kind == SlbT_WATER &&  computer_check_room_of_role_available(comp, RoRoF_PassWater) == IAvail_Now)||
```

Comment:
Here the slab_kind_is_liquid can be removed too.

Reply:
done